packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -223,6 +274,23 @@ export function ClusterServer<F extends BitSchema, SF extends TypeFromBitSchema<
     const generatedCommandList = new Array<number>();
     for (const name in commandDef) {
         const handler = (handlers as any)[name];
+
+        if (isConditionalCommand(commandDef[name])) {
+            const { mandatoryIf, optionalIf } = commandDef[name] as any;
```

Comment:
If above test was commandDef[name].isConditional then it would act as a type assertion and the cast to any would not be nessary.  If not that, then isConditionalCommand could assert type to make cast unnecessary.

Reply:
done